### PR TITLE
Fix hack/test-integration.sh on OSX

### PR DESCRIPTION
**Pull request by liggitt**
First commit just fixes typos and invocations that don't exist on OSX

Second commit removes a workaround that breaks building the test binary (it doesn't actually contain any of the integration tests) on my combination of go (1.7.5), OSX (10.12.5) and clang (Apple LLVM version 8.1.0 (clang-802.0.42), Target: x86_64-apple-darwin16.6.0)

### CODE REVIEW

**Comment by liggitt** [comment]
@smarterclayton do you know what the code signing workaround was fixing? I want to make sure it is no longer required, or gate it on a go version or OS version or clang version

**Comment by smarterclayton** [comment]
Are ou on the latest Xcode?

**Comment by liggitt** [comment]
I don't have xcode installed, I have the developer tools installed, and am at the latest version. what do you see for these commands:

```sh
$ xcode-select -p
/Library/Developer/CommandLineTools

$ xcode-select --install
xcode-select: error: command line tools are already installed, use "Software Update" to install updates

$ gcc --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Comment by smarterclayton** [comment]
$ xcode-select -p
/Applications/Xcode.app/Contents/Developer

$ xcode-select --install

not installed

$ gcc --version
Configured with:
--prefix=/Applications/Xcode.app/Contents/Developer/usr
--with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

$ ld -v
@(#)PROGRAM:ld  PROJECT:ld64-278.4
configured to support archs: armv6 armv7 armv7s arm64 i386 x86_64
x86_64h armv6m armv7k armv7m armv7em (tvOS)
LTO support using: LLVM version 8.1.0, (clang-802.0.42)
TAPI support using: Apple TAPI version 1.33.11


​I still periodically get the error on 1.7.5

**Comment by liggitt** [comment]
What version of OSX are you running?

> ​I still periodically get the error on 1.7.5

What is the error you are seeing, just a `make` and run, or some separate code-signing step?

**Comment by smarterclayton** [comment]
The error is during linking - without omitting dwarf info a certain size of
binary (usually openshift / extended.test) fails with the exit code 9

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
reworked to remove reliance on symbols in the test binary, just pulled out test names with `go list` directly. @smarterclayton PTAL

**Comment by smarterclayton** [comment]
Oh yeah, go list.  That's smart.

**Comment by smarterclayton** [comment]
Lgtm [merge]

**Comment by smarterclayton** [comment]
[severity:blocker] tests only

File: hack/test-integration.sh [hunk 1]
```diff
@@ -121,10 +127,10 @@ for test in "${tests[@]}"; do
 		fi
 	done
 done
-test_end_time=$(date +%s%3N)
+test_end_time=$(date +%s)
```
Comment (stevekuznetsov) [inline hack/test-integration.sh]: Isn't this formatting needed for jUnit? Can we confirm it is being generated appropriately now?

Comment (liggitt) [inline hack/test-integration.sh]: Inserted milliseconds below

**Pull request merged**